### DISCUSSION TITLE
CMake config is not included

### DISCUSSION THREAD

**Discussion by ianhbell** [wjakob/nanobind#777] [General]
I am building a nanobind interface for nlohmann::json, the nanobind variant of [pybind11_json](https://github.com/pybind/pybind11_json). I cannot use all their tooling directly because the CMake config is not a part of the nanobind conda package. For pybind11 we have the necessary file here:
```
/Users/ihb/mambaforge/envs/py312/lib/python3.12/site-packages/pybind11/share
├── cmake
│   └── pybind11
│       ├── FindPythonLibsNew.cmake
│       ├── pybind11Common.cmake
│       ├── pybind11Config.cmake
│       ├── pybind11ConfigVersion.cmake
│       ├── pybind11GuessPythonExtSuffix.cmake
│       ├── pybind11NewTools.cmake
│       ├── pybind11Targets.cmake
│       └── pybind11Tools.cmake
└── pkgconfig
    └── pybind11.pc
```

but the same ``nanobindConfig.cmake`` file does not exist for nanobind. Can you please add it?

**Comment by ianhbell** [comment]
There is a nanobind-config.cmake:
```
/Users/ihb/mambaforge/envs/py312/lib/python3.12/site-packages/nanobind/cmake
├── darwin-ld-cpython.sym
├── darwin-ld-pypy.sym
├── nanobind-config-version.cmake
└── nanobind-config.cmake
```
but CMake does not find it, even when this package is installed via conda

**Reply by ianhbell** [reply]
I was able to force CMake to find the module with this cludge but it is not a cross-platform solution:
```
-D nanobind_DIR=`python -c "import nanobind, os; print(os.path.dirname(nanobind.__file__)+'/cmake')"`

```
I think possibly the same folder structure is needed, but I don't know the logic of how conda and cmake interact

**Reply by wojdyr** [reply]
Use `python -m nanobind --cmake_dir`.
Alternatively, you could try scikit-build-core, which handles it automatically.

**Reply by ianhbell** [reply]
I see - is that documented somewhere? scikit-build-core does not get invoked when building the tests

**Reply by ianhbell** [reply]
It would be convenient if no magical incantations would be required to get nanobind stuff pulled in to minimize migration headaches

**Reply by wjakob** [reply]
I don't work with Conda myself and just do the minimal version bumps to keep the package in sync. Note that the `python -m nanobind --cmake_dir` incantation is actually the official way to get nanobind's CMake include directory, and this is also documented. But it could be the case that it only includes `nanobind-config.cmake` while lacking some auto-generated CMake files. If you have suggestions and improvements, it would be great if you can open a PR on the feedstock.

**Reply by ianhbell** [reply]
Where is the feedstock? I think it is just a question of folder structure if I understand correctly. I can see about fixing it.

**Reply by wjakob** [reply]
in the conda-forge master repository like all of the other feedstocks.

**Comment by RaulPPelaez** [comment]
For the record, CMake's FetchContent seems to work fine:
```cmake
include(FetchContent)
FetchContent_Declare(
  nanobind
  GIT_REPOSITORY https://github.com/wjakob/nanobind/
  GIT_TAG        v2.4.0
)
FetchContent_MakeAvailable(nanobind)
```
